### ISSUE TITLE
expirationDateTime option is not working while adding the permission

### ISSUE THREAD

**Issue by vikasprabhakar** [microsoftgraph/microsoft-graph-docs#8820]
[Enter feedback here]
expirationDateTime option is not working while adding the permission
Request: https://graph.microsoft.com/v1.0/sites/<path>test:/microsoft.graph.invite
Method : POST
Body: {
    "requireSignIn": true,
    "roles": [
        "read"
    ],
    "sendInvitation": false,
    "message": "Here's the file that we're collaborating on.",
    "recipients": [
        {
            "email": "vikas****@*****.com"
        }
    ],
    "expirationDateTime": "2020-06-24T11:13:54Z"
}
It is giving the errors:
{
  "error": {
    "code": "-1, Microsoft.SharePoint.Client.InvalidClientQueryException",
    "message": "The parameter expirationDateTime does not exist in method invite.",
    "innerError": {
      "date": "2020-06-23T04:30:25",
      "request-id": "6e5a589c-a99e-400d-ad25-d98e25d53350"
    }
  }
}

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b75e1562-24d6-bfaf-be46-d52ee93ffb1d
* Version Independent ID: cd35f8f6-774e-70d3-9bd5-11110ab450b1
* Content: [Send an invite to access an item - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/driveitem-invite?view=graph-rest-1.0&tabs=http)
* Content Source: [api-reference/v1.0/api/driveitem-invite.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/v1.0/api/driveitem-invite.md)
* Product: **sharepoint**
* Technology: **microsoft-graph**
* GitHub Login: @JeremyKelley
* Microsoft Alias: **JeremyKelley**

**Comment by vikasprabhakar** [comment]
Hi @JeremyKelley , please provide your feedbacks on the above issue.

**Comment by Sanjay12089** [comment]
Hi @vikasprabhakar  ,

Did you find any solution for this issue?
I am also getting same error.

**Comment by shrirajgohil** [comment]
Hi @JeremyKelley ,

Please provide some insights for this issue as we are still facing the same.

Thanks

**Comment by vikasprabhakar** [comment]
> Hi @vikasprabhakar ,
> 
> Did you find any solution for this issue?
> I am also getting same error.

Hi @Sanjay12089 ,
I didn't get any solution till now.
I am still waiting @JeremyKelley inputs on it.

**Comment by shrirajgohil** [comment]
> > Hi @vikasprabhakar ,
> > Did you find any solution for this issue?
> > I am also getting same error.
> 
> Hi @Sanjay12089 ,
> I didn't get any solution till now.
> I am still waiting @JeremyKelley inputs on it.

Hi @vikasprabhakar,

We had this resolved by removing expirationDateTime and password from the methods parameter.

Just like this  :- "_var DestinationPermission = await graphServiceClient.Groups[team ID].Drive.Items[File ID].Invite(recipients, requireSignIn, roles, sendInvitation, message).Request().PostAsync();_"

Hope this helps!

**Comment by vikasprabhakar** [comment]
> > > Hi @vikasprabhakar ,
> > > Did you find any solution for this issue?
> > > I am also getting same error.
> > 
> > 
> > Hi @Sanjay12089 ,
> > I didn't get any solution till now.
> > I am still waiting @JeremyKelley inputs on it.
> 
> Hi @vikasprabhakar,
> 
> We had this resolved by removing expirationDateTime and password from the methods parameter.
> 
> Just like this :- "**_var DestinationPermission = await graphServiceClient.Groups[team ID].Drive.Items[File ID].Invite(recipients, requireSignIn, roles, sendInvitation, message).Request().PostAsync();_**"
> 
> Hope this helps!

Hi @Sanjay12089 ,
Yes agree, the API is working without expirationDateTime and password options.
I have a requirements of give permission to some time extent thats why exploring the "expirationDateTime".
Unfortunately the API is not working with "expirationDateTime" option.

**Comment by js-ojus** [comment]
I have the same issue.

@JeremyKelley : Could you please provide some clarification / help?  Thanks.

**Comment by js-ojus** [comment]
@JeremyKelley or any other Microsoft team member!

This is clearly a bug, since the observed behaviour is at variance with that which is promised by the official documentation.

We have performed an integration with OneDrive as part of a critical workflow, and now this is not working at the very last stage of the workflow.  It is a huge inconvenience (bordering on impractical), should we have to manually remember and remove each individual share that was given as part of the automated workflow.

Sorry for being a pest, but it helps a lot if Microsoft could provide *some* response at all.

**Comment by vikasprabhakar** [comment]
No response from Microsoft team in last one month. This is not expected from Microsoft.
@JeremyKelley or any Microsoft team member, plz look the issue and provide some response.

**Comment by christopher-walker** [comment]
Also having the same issue with adding the expire date time.
There are other parts of this API which are not working as expected either, for example when you set "sendInvitation" to true, the users are added to the sharepoint list i.e. breaking role inheritance, when you create a sharing link this should not happen.

`sendInvitation | Boolean | If true, a sharing link is sent to the recipient. Otherwise, a permission is granted directly without sending a notification.`

So in summary I think this is not working as users are granted permission directly to the drive item (in my case in SharePoint) this type of sharing does not support expirationDateTime therefor this also fails.

Seems to be the fault of the API.

**Comment by ssapa** [comment]
I have the same issue.

Could you please provide some clarification / help? Thanks.

**Comment by christopher-walker** [comment]
Custom applications using this API could be successfully failing i.e. sharing items but not using the modern sharing link functionality. As a result customers could be approching and or exceding the unuque permissions threashold for document libraries in SharePoint without realising that an issue is pressent:

https://docs.microsoft.com/en-us/sharepoint/troubleshoot/lists-and-libraries/error-share-break-inheritance#:~:text=The%20supported%20limit%20of%20unique,at%20a%20time%20takes%20longer.

The threashold of 5000 is not massive either and can easily be hit with a small number of documents or folders shared with a small number of people. Therefor it is not out of the imagination that the supported limit of 50,000 unique permissions could be hit fairly quickly for anything other than a small list.

Once discovered there could be a very large cleanup job to resolve this in customers environments. This really needs to be looked at and addressed.

**Comment by michwii** [comment]
Hello,

I have the same problem.
Any news about the subject ?

Regards
